addons/purchase/models/purchase.py

```diff
@@ -86,13 +86,18 @@ def _compute_picking(self):
                 # do some recursive search, but that could be prohibitive if not done correctly.
                 moves = line.move_ids | line.move_ids.mapped('returned_move_ids')
                 pickings |= moves.mapped('picking_id')
+            # show all shipment related to current purchase order
+            if order.group_id:
+                pickings |= self.env['stock.picking'].search([('group_id', '=', order.group_id.id)])
```

Comment:
in line 87 and 88 should we not only look for returned move? i guess this search will also gives back the first receipt

maybe you can add picking_ids o2m on procurement group to be cleaner?